### PR TITLE
Dev 230/feature flag integration

### PR DESCRIPTION
**Pull request by aaronipollock**
Flags should refresh every time the app is opened and not hit the API on every request.

## Summary by Sourcery

Integrate a feature flag system using a new FeatureFlagProvider to manage and refresh feature flags efficiently. Enhance the application with a new ApolloClient setup for GraphQL requests and introduce a TypeScript configuration file for better build management.

New Features:
- Introduce a FeatureFlagProvider to manage feature flags within the application, allowing flags to be refreshed when the app is opened.

Enhancements:
- Add a new ApolloClient configuration to manage GraphQL requests with caching capabilities.

Build:
- Add a tsconfig.json file extending the base configuration from Expo.

### CODE REVIEW

File: libs/expo/shared/clients/src/lib/apollo/apolloClient.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+import { ApolloClient, InMemoryCache } from "@apollo/client";
+
+const client = new ApolloClient({
+    uri: process.env.REACT_APP_GRAPHQL_ENDPOINT || "https://dev.api.example.com/graphql",
```
[the remark this answers is not in the document]
Comment (aaronipollock) [reply libs/expo/shared/clients/src/lib/apollo/apolloClient.ts:5]: Deleted file

File: libs/expo/betterangels/src/lib/providers/featureFlag/featureFlagProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    if (loading) {
+        return (
+            <View>
+                <Text>Loading...</Text>
+            </View>
+        );
+    }
+
+    if (error) {
+        console.error("FeatureFlagProvider Error:", error);
```
[the remark this answers is not in the document]
Comment (aaronipollock) [reply libs/expo/betterangels/src/lib/providers/featureFlag/featureFlagProvider.tsx:50]: Refactored:

    if (error) {
        console.error("FeatureFlagProvider encountered an error:", error.message);
        return (
            <View>
                <Text>Something went wrong. Please try again later.</Text>
            </View>
        );
    }

File: .python-version [hunk 3]
```diff
@@ -1 +1 @@
-3.13.0
+3.12.4
```
Comment (vecchp) [inline .python-version:1]: why are we downgrading the python version here?

File: libs/expo/betterangels/src/lib/screens/AddEditClient/index.tsx [hunk 4]
```diff
@@ -378,6 +378,7 @@ export default function AddEditClient({ id }: { id?: string }) {
             )}
           </MainScrollContainer>
           <BottomActions
+            submitTitle="Update"
```
Comment (vecchp) [inline libs/expo/betterangels/src/lib/screens/AddEditClient/index.tsx:381]: What is this change in relation to this PR?

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+  "compilerOptions": {},
+  "extends": "expo/tsconfig.base"
+}
```
Comment (vecchp) [inline tsconfig.json:4]: Is this supposed to be here?

File: libs/expo/betterangels/src/lib/screens/UserProfile/index.tsx [hunk 6]
```diff
@@ -70,6 +70,7 @@ export default function UserProfile() {
         {userInfo.map((item, index) => (
           <InfoCard key={index} title={item.title} value={item.value} />
         ))}
+
```
Comment (vecchp) [inline libs/expo/betterangels/src/lib/screens/UserProfile/index.tsx:73]: Any reason why we are updating this file?

File: libs/expo/betterangels/src/lib/providers/featureFlag/featureFlagProvider.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    }
+`;
+
+interface FeatureFlagProviderProps {
+    children: React.ReactNode;
+}
+
+export const FeatureFlagProvider: React.FC<FeatureFlagProviderProps> = ({ children }) => {
+    const [featureFlags, setFeatureFlags] = useState<FeatureFlags>({});
+    const { data, loading, error } = useQuery(FEATURE_FLAGS_QUERY);
```
Comment (vecchp) [inline libs/expo/betterangels/src/lib/providers/featureFlag/featureFlagProvider.tsx:24]: We can use an autogenerated hook here, so there's no need to pass in the query manually. This approach also provides a clean return type for the query. You can see an example here: [UserProvider.tsx](https://github.com/BetterAngelsLA/monorepo/blob/f6d0f9e049578d3761c8d8dd5d7dccae6780ddfa/libs/expo/betterangels/src/lib/providers/user/UserProvider.tsx#L15-L17).

Running ynx-generate within the dev container should generate the necessary types for you.

**Review by vecchp — CHANGES_REQUESTED** [review]
Overall, excellent work on implementing the feature flag integration! Addressing the above points will help get this PR in to an even better state. Keep up the good work!

**Comment by aaronipollock** [comment]
@vecchp This PR is ready for your review again.

File: .python-version [hunk 8]
```diff
@@ -1 +1 @@
-3.13.0
+3.10.0
```
Comment (tglaz) [inline .python-version:1]: any reason to downgrade?

File: libs/expo/betterangels/src/lib/providers/featureFlag/featureFlagProvider.tsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+interface FeatureFlagProviderProps {
+    children: React.ReactNode;
+}
+
+export const FeatureFlagProvider: React.FC<FeatureFlagProviderProps> = ({ children }) => {
+    const [featureFlags, setFeatureFlags] = useState<FeatureFlags>({});
+    const { data, error } = useGetFeatureFlagsQuery();
+
+    useEffect(() => {
+        if (data?.featureControls?.flags) {
```
Comment (tglaz) [inline libs/expo/betterangels/src/lib/providers/featureFlag/featureFlagProvider.tsx:15]: perhaps we should reset featureFlags if there's no data to be safe.
so something like this:
```javascript
if (!data?.featureControls?.flags) {
   setFeatureFlags({});

   return 
}

const flags = ....

```

**Review by tglaz — CHANGES_REQUESTED** [review]

**Comment by vecchp** [comment]
merged in https://github.com/BetterAngelsLA/monorepo/pull/779

**Pull request closed without merging**